### ISSUE TITLE
In method detectAndCompute parameter 4 is not homogeneous for descriptor (ORB BRISK KAZE AKAZE)

### ISSUE THREAD

**Issue by LaurentBerger** [opencv/opencv#5196]
In this small program method detectAndCompute send an exception for only Akaze and Kaze descriptor when parameter 4  descriptor is of type UMat. This does not occur for Orb and Brisk.

``` .cpp
    UMat m;
    imread("f:/lib/opencv/samples/data/lena.jpg",CV_LOAD_IMAGE_ANYCOLOR).copyTo(m);
    cout << "Original image\n";
    UMat masqueMat=UMat();
    UMat descUMat;
    Mat descMat;
    vector<KeyPoint> kpts1;
    Ptr<cv::Feature2D> akaze;
    akaze = AKAZE::create();
    Ptr<cv::Feature2D> kaze;
    kaze = KAZE::create(); // Error in first code change AKAZE in KAZE same exception
    Ptr<cv::Feature2D> orb;
    orb = ORB::create();
    Ptr<cv::Feature2D> brisk;
    brisk = BRISK::create();

    orb->detectAndCompute(m,masqueMat, kpts1, descUMat); //OK with Umat
    brisk->detectAndCompute(m,masqueMat, kpts1, descUMat); // OK with Umat
    kpts1.clear();
    descUMat = UMat();
    akaze->detectAndCompute(m,masqueMat, kpts1, descMat); // OK with mat
    kaze->detectAndCompute(m,masqueMat, kpts1, descMat); // OK with mat
    try
    {
        kaze->detectAndCompute(m,masqueMat, kpts1, descUMat);// EXCEPTION with Umat
    }
    catch(cv::Exception& e)
        {
        cout  <<"KAZE with UMat descriptor "<<e.msg<<"\n";
        }
    try
    {
        akaze->detectAndCompute(m,masqueMat, kpts1, descUMat);// EXCEPTION with Umat
    }
    catch(cv::Exception& e)
        {
        cout  <<"AKAZE with UMat descriptor "  <<e.msg<<"\n";
        }
```

**Comment by catree** [comment]
@LaurentBerger 
Looks like the bug cannot be reproduced anymore. Maybe you can confirm?
Probably solved by #9330 and related pull requests.

**Comment by LaurentBerger** [comment]
@catree Just try no exception Ok to close

opencv version

General configuration for OpenCV 4.0.0-pre =====================================
  Version control:               3.4.1-1801-g220b27857
